File: 01_big_O_notation/05_rules/02_sum_rule/sum_rule.py
"""
 f(n) = O(p(n))
                ->  O(p(n) + q(n))
 g(n) = O(q(n))
"""


def f(n):
    accum = 0
    for i in range(n):
        accum += 1
    return accum


def g(n):
    accum = 0
    c = 10
    for i in range(c * n):
        accum += 1

    return accum


def sum_fg(n):
    f(n)
    g(n)
    return 0


def sum_fg_v2(n):
    accum = 0
    c = 10

    for i in range(n):
        accum += 1

    for i in range(c * n):
        accum += 1

    return accum


if __name__ == '__main__':
    n = 5
    print(sum_fg(n))
    print(sum_fg_v2(n))

File: 01_big_O_notation/05_rules/02_sum_rule/sum_rule.js
/*
 f(n) = O(p(n))
                ->  O(p(n) + q(n))
 g(n) = O(q(n))
*/

function f(n) {
    let accum = 0
    for (let i = 0; i < n; i ++) {
        accum +=1
    }    
    return accum
}

function g(n) {
    let accum = 0
    let c = 10
    for (let i = 0; i < c * n; i ++) {
        accum += 1
    }
    return accum
}

function sum_fg(n) {
    f(n)
    g(n)
    return 0
}

function sum_fg_v2(n) {
    let accum = 0
    let c = 10

    for (let i = 0; i < n; i ++) {
        accum += 1
    }

    for (let i = 0; i < c * n;  i ++) {
        accum +=1
    }

    return accum

}

n = 5
console.log( sum_fg(n) )
console.log( sum_fg_v2(n) )
